### PR TITLE
Wmg/13 login

### PR DESCRIPTION
**Pull request by allthesignals**
This PR adds ember-simple-auth, using the devise authenticator. It also integrates with mirage, which responds with the correct JSON object (as opposed to headers). 

Struggling with the acceptance test, found commented below. I commented out what I was trying, but it broke the test suite and kept trying to redirect the test suite itself to the `/login` route (which said "Cannot find route /login". This probably has something to do with the click helpers.

### CODE REVIEW

File: tests/acceptance/user-logs-in-and-logs-out-successfully-test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  beforeEach(function() {
+    application = startApp();
+  });
+
+  afterEach(function() {
+    destroyApp(application);
+  });
+
+  // describe('show index', function() {
```
Comment (allthesignals) [inline tests/acceptance/user-logs-in-and-logs-out-successfully-test.js]: @cball here are the commented tests I mentioned in the PR above.

File: app/authenticators/devise.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import DeviseAuthenticator from 'ember-simple-auth/authenticators/devise';
+
+export default DeviseAuthenticator.extend({
+  serverTokenEndpoint: 'api/token'
```
Comment (cball) [inline app/authenticators/devise.js]: this needs to be `/api/token`

File: app/templates/application.hbs [hunk 3]
```diff
@@ -8,7 +8,11 @@
       </div>
     {{/each}}
   <div class="menu">
-    Login
+    {{#if session.isAuthenticated}}
+      <a {{action 'invalidateSession'}}>Logout</a>
+    {{else}}
+      {{#link-to 'login'}}Login{{/link-to}}
```
Comment (cball) [inline app/templates/application.hbs]: I'd use the inline version here:

```hbs
{{link-to 'Login' 'login'}}
```

File: tests/acceptance/user-logs-in-and-logs-out-successfully-test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  //   //     // click(LOGIN_LINK);
+  //   //     fillIn(EMAIL_FIELD, 'user@email.com');
+  //   //     fillIn(PASSWORD_FIELD, 'user@email.com');
+  //   //   });
+
+  //   //   it("shows login boxes", function() {
+  //   //     // expect(currentURL()).to.equal("/login");
+  //   //   });
+
+  //   //   andThen(function() {
```
Comment (cball) [inline tests/acceptance/user-logs-in-and-logs-out-successfully-test.js]: this was the major bug in the test. `andThen` needs to be inside of a `beforeEach` or `it` block. This was causing an exception, which was hitting this line in the authenticator: https://github.com/simplabs/ember-simple-auth/blob/master/addon/authenticators/devise.js#L159

**Comment by cball** [comment]
@allthesignals could you enable protection for the `develop` branch and set it to allow merge only if CI passes and review is 👍 ?

I know for me if I see a bunch of green I just merge it 😄

**Comment by allthesignals** [comment]
@cball Alright, I addressed requested changes, then went ahead and compared what I did with your version of this branch. The commits were basically the same - I think in the test, though, you ended up avoiding `andThen`. Maybe it was unnecessary when I was trying it. Anyway, I pulled your changes in and prioritized them. 

There are a bunch of weird duplicate commits here for some reason because I needed to rebase the branch with develop. I'm not sure why that happened. 

I also made the protected branches and required 👍  review! Thanks for the input. Moving on to other reviews.

**Comment by allthesignals** [comment]
Alright, because other branches depend on this one, and it's causing some issues with tests, I'm going to go ahead and merge this. If you had anymore specific issues with stuff here, let me know in comments and I can make an issue. Thanks!

**Pull request merged**